### PR TITLE
py3(django): pytest startup and test collection on Django 1.10

### PR DESCRIPTION
**Pull request by joshuarli**
At this point, pytest should start and successfully collect all tests.

Note that this is lowest common denominator of changes. For example, there are some issues with `get_all_field_names` but not all of them block pytest. So we don't need to rebase on things like https://github.com/getsentry/sentry/pull/15647.

### CODE REVIEW

**Comment by wedamija** [comment]
Might be worth a separate pr for the changes to `urlpatterns` since that change is pretty large, and should be compatible with 1.9 as well.

**Comment by joshuarli** [comment]
I double checked to see whether I was doing https://docs.djangoproject.com/en/1.10/ref/templates/upgrading/#the-templates-settings right by printing `settings.TEMPLATES` under Django 1.9. (It's been deprecated since 1.8, and Django autogenerates `settings.TEMPLATES`:

```
[{'APP_DIRS': True,
  'BACKEND': 'django.template.backends.django.DjangoTemplates',
  'DIRS': ['/Users/joshua.li/dev/sentry/sentry/src/sentry/templates'],
  'OPTIONS': {'context_processors': ['django.contrib.auth.context_processors.auth',
                                     'django.contrib.messages.context_processors.messages',
                                     'django.template.context_processors.csrf',
                                     'django.template.context_processors.request',
                                     'social_auth.context_processors.social_auth_by_name_backends',
                                     'social_auth.context_processors.social_auth_backends',
                                     'social_auth.context_processors.social_auth_by_type_backends',
                                     'social_auth.context_processors.social_auth_login_redirect']}}]
```

File: src/sentry/new_migrations/monkey/__init__.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 from django.db.migrations import migration, executor, writer
 
-LAST_VERIFIED_DJANGO_VERSION = (1, 9)
+LAST_VERIFIED_DJANGO_VERSION = (1, 10)
```
Comment (joshuarli) [inline src/sentry/new_migrations/monkey/__init__.py]: Gonna revert this once approved.

**Comment by joshuarli** [comment]
And here's a verification for `LAST_VERIFIED_DJANGO_VERSION`.

Looks like the migration template didn't change in Django 1.10, so no changes required to `SENTRY_MIGRATION_TEMPLATE`.

-- WIP --

**Review by wedamija — APPROVED** [review]

**Pull request merged**